straxen/url_config.py

```diff
@@ -329,3 +334,66 @@ def open_neural_net(model_path: str, **kwargs):
         tar = tarfile.open(model_path, mode="r:gz")
         tar.extractall(path=tmpdirname)
         return tf.keras.models.load_model(tmpdirname)
+
+
+@URLConfig.register('itp_dict')
+def get_itp_dict(loaded_json,
+                 run_id=None,
+                 time_key='time',
+                 itp_keys='correction',
+                 **kwargs) -> typing.Union[np.ndarray, typing.Dict[str, np.ndarray]]:
+    """
+    Interpolate the dictionary at the start time that is queried from
+    a run-id.
+
+    The loaded JSON (dictionary), should have at least one collum `time`
+    and has other columns given by `itp_dict_keys`.
+
+    :param loaded_json: a dictionary with a time-series
+    :param run_id: run_id
+    :param time_key: key that gives the timestamps
+    :param itp_keys: which keys from the dict to read. Should be
+        comma (',') separated!
+
+    :return: Interpolated values of dict at the start time, either
+        returned as an np.ndarray (single value) or as a dict
+        (multiple itp_dict_keys)
+    """
+    keys = strax.to_str_tuple(itp_keys.split(','))
+    times = loaded_json[time_key]
+    if not all(k in loaded_json for k in keys):
+        raise ValueError(f'One or more of {itp_keys} are not in {loaded_json.keys()}')
+
+    # get start time of this run. Need to make tz-aware
+    start = xent_collection().find_one({'number': int(run_id)}, {'start': 1})['start']
+    start = pytz.utc.localize(start).timestamp() * 1e9
+
+    try:
+        if len(strax.to_str_tuple(keys))>1:
+            return {key:
+                    interp1d(times, loaded_json[key], bounds_error=True)(start)
+                    for key in keys}
+
+        else:
+            interp = interp1d(times, loaded_json[keys[0]], bounds_error=True)
+            return interp(start)
+    except ValueError as e:
+        raise ValueError(f"The correction is not defined for run {run_id}") from e
+
+
+@URLConfig.register('rekey_dict')
+def rekey_dict(d, replace_keys='', with_keys=''):
+    '''
+    :param d: dictionary that will have its keys renamed
+    :param replace_keys: comma-separated list of keys that will be replaced
+    :param with_keys:  comma-separated list of keys that will replace the replace_keys
```

Comment:
Comma separated list reads somewhat funny. I see what you mean, but 'comma-separated' string might be more intuitive idk.

Reply:
Done